internal/testutil/datagen/datagen.sh

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env bash
+
+set -e
+
+# scrapper_selector is the label with which the scraper deployment is deployed.
+scrapper_selector="app=scraper"
+
+# main subcommand runs the whole flow of the script: Bootstrap, scrape, and cleanup
+function main() {
+    if [[ -z "$1" ]]; then
+        echo "Usage: $0 <output_folder>"
+        exit 1
+    fi
+
+    bootstrap
+
+    pod=$(scraper_pod "$scrapper_selector")
+    if [[ -z "pod" ]]; then
+        echo "Could not find scraper pod (-l $scrapper_selector)"
+        exit 1
+    fi
+
+    mkdir -p "$1" && cd "$1"
+    testinfo > README.md
+
+    # Kubelet endpoints
+    mkdir -p kubelet
+    echo "Extracting kubelet /pods"
+    kubectl exec -n scraper $pod -- datagen.sh scrape kubelet pods > kubelet/pods
+
+    echo "Extracting kubelet /metrics/cadvisor"
+    mkdir -p kubelet/metrics
+    kubectl exec -n scraper $pod -- datagen.sh scrape kubelet metrics/cadvisor > kubelet/metrics/cadvisor
+
+    echo "Extracting kubelet /stats/summary"
+    mkdir -p kubelet/stats
+    kubectl exec -n scraper $pod -- datagen.sh scrape kubelet stats/summary > kubelet/stats/summary
+
+    # KSM endpoint
+    mkdir -p ksm
+    echo "Extracting ksm /metrics"
+    kubectl exec -n scraper $pod -- datagen.sh scrape ksm > ksm/metrics
+
+    if [[ "$DISABLE_CONTROLPLANE" != "1" ]]; then
+        # Control plane components
+        mkdir -p controlplane/api-server
+        echo "Extracting api-server /metrics"
+        kubectl exec -n scraper $pod -- datagen.sh scrape controlplane apiserver > controlplane/api-server/metrics
+
+        mkdir -p controlplane/etcd
+        echo "Extracting etcd /metrics"
+        kubectl exec -n scraper $pod -- datagen.sh scrape controlplane etcd > controlplane/etcd/metrics
+
+        mkdir -p controlplane/controller-manager
+        echo "Extracting controller-manager /metrics"
+        kubectl exec -n scraper $pod -- datagen.sh scrape controlplane controllermanager > controlplane/controller-manager/metrics
+
+        mkdir -p controlplane/scheduler
+        echo "Extracting scheduler /metrics"
+        kubectl exec -n scraper $pod -- datagen.sh scrape controlplane scheduler > controlplane/scheduler/metrics
+    else
+        echo "Skipping control plane metrics"
+    fi
+
+    # K8s objects
+    echo "Generating list of kubernetes resources"
+    kubedump namespaces
+    kubedump services
+    kubedump pods
```

Comment:
We need to add more here, at least `endpoints`.

Reply:
also nodes

Reply:
Done in https://github.com/newrelic/nri-kubernetes/pull/269/commits/1d8bb1f6a3d21f17dc16e9e4f0fa0e0ea373a6c1